src/QueryRenderer.ts

```diff
@@ -206,6 +211,39 @@ class QueryRenderChild extends MarkdownRenderChild {
         });
     }
 
+    private static addGroupHeadings(
+        content: HTMLDivElement,
+        groupHeadings: GroupHeading[],
+    ) {
+        for (const heading of groupHeadings) {
+            QueryRenderChild.addGroupHeading(content, heading);
+        }
+    }
+
+    private static addGroupHeading(
+        content: HTMLDivElement,
+        group: GroupHeading,
+    ) {
+        const groupName = group.title;
+        const level = group.level;
+        let header: any;
```

Comment:
A concrete type would be good.

Reply:
The concrete type is `GroupHeading` - on line 225.

Happy to inline those variables and repeat group.level in the `if` block.

Really the way that whole functions calculates h4, h5 or h6 is horrible... Am hoping someone more knowledgeable about TS can collapse that whole `if` block down to a one-liner....

Reply:
Fixed.